operations/update.go

```diff
@@ -53,96 +59,116 @@ func Update() cli.Command {
 			if err != nil {
 				return errors.Wrap(err, "problem loading configuration")
 			}
-
-			client := conf.getRestCommunicator(ctx)
-			defer client.Close()
-
-			update, err := checkUpdate(client, false, forceUpdate)
-			if err != nil {
-				return err
+			if !conf.AutoUpgradeCLI && !autoUpgrade {
+				fmt.Println(fmt.Sprintf("Automatic CLI upgrades are not set up; specifying the -%s flag will enable automatic CLI upgrades before each command if the current CLI is out of date.", autoUpgradeFlagName))
 			}
-			if !update.needsUpdate || len(update.binaries) == 0 {
-				return nil
+			if conf.AutoUpgradeCLI && autoUpgrade {
+				fmt.Println(fmt.Sprintf("Automatic CLI upgrades are already set up, specifying the %s flag is not necessary.", autoUpgradeFlagName))
 			}
-
-			updatedBin, err := tryAllPrepareUpdate(update)
-			if err != nil {
-				return err
-			}
-			defer func() {
-				if doInstall {
-					grip.Error(os.Remove(updatedBin))
+			if !conf.AutoUpgradeCLI && autoUpgrade {
+				conf.SetAutoUpgradeCLI()
+				if err := conf.Write(""); err != nil {
+					return errors.Wrap(err, "error setting auto-upgrade CLI option")
```

Comment:
what does conf.Write("") do?

Reply:
When passing an empty string, the conf struct will be written to the default config file location stored in conf.LoadedFrom. If that is also not set, an error gets thrown.